compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -2197,24 +2205,35 @@ impl<'a, 'tcx> InferCtxtPrivExt<'a, 'tcx> for InferCtxt<'a, 'tcx> {
     fn annotate_source_of_ambiguity(
         &self,
         err: &mut Diagnostic,
-        impls: &[DefId],
+        impls: &[AmbiguousSelection],
         predicate: ty::Predicate<'tcx>,
     ) {
         let mut spans = vec![];
         let mut crates = vec![];
         let mut post = vec![];
-        for def_id in impls {
-            match self.tcx.span_of_impl(*def_id) {
-                Ok(span) => spans.push(self.tcx.sess.source_map().guess_head_span(span)),
-                Err(name) => {
-                    crates.push(name);
-                    if let Some(header) = to_pretty_impl_header(self.tcx, *def_id) {
-                        post.push(header);
+        let mut or_where_clause = false;
+        for ambig in impls {
+            match ambig {
+                AmbiguousSelection::Impl(def_id) => match self.tcx.span_of_impl(*def_id) {
+                    Ok(span) => spans.push(self.tcx.sess.source_map().guess_head_span(span)),
```

Comment:
```suggestion
                    Ok(span) => spans.push(span),
```
You shouldn't need `guess_head_span` any more.

Reply:
Why isn't it needed anymore?

Reply:
@michaelwoerister, cjgillot changed the `def_span` (and thus `span_of_impl`) to point to the head span by default, so `guess_head_span` is redundant.